### PR TITLE
Put back identityfile parameter from .ssh/config

### LINKED ISSUE #2845: Bring back support OpenSSH Identityfile
# Description

Support for identity file was added in - #2534
And then removed during refactoring in - #2691 

# Proposal

Add it back :) in PR #2846

### General Info

```bash
❯❯❯ docker --version
Docker version 20.10.6, build 370c289

❯❯❯ docker-compose version
docker-compose version 1.29.2, build 5becea4c
docker-py version: 5.0.0
CPython version: 3.9.0
OpenSSL version: OpenSSL 1.1.1h  22 Sep 2020
```
MacOS 11.3.1

### PR DESCRIPTION
**Pull request by shehzaman**
Signed-off-by: Shehzaman

fixes #2845

### CODE REVIEW

File: docker/transport/sshconn.py [hunk 1]
```diff
@@ -205,17 +205,18 @@ def _create_paramiko_client(self, base_url):
             with open(ssh_config_file) as f:
                 conf.parse(f)
             host_config = conf.lookup(base_url.hostname)
-            self.ssh_conf = host_config
```
Comment (shehzaman) [inline docker/transport/sshconn.py:208]: Removed this variable, it was contributing nothing to the code.

File: docker/transport/sshconn.py [hunk 2]
```diff
@@ -205,17 +205,18 @@ def _create_paramiko_client(self, base_url):
...
             if 'hostname' in host_config:
                 self.ssh_params['hostname'] = host_config['hostname']
             if base_url.port is None and 'port' in host_config:
-                self.ssh_params['port'] = self.ssh_conf['port']
+                self.ssh_params['port'] = host_config['port']
             if base_url.username is None and 'user' in host_config:
-                self.ssh_params['username'] = self.ssh_conf['user']
+                self.ssh_params['username'] = host_config['user']
+            if 'identityfile' in host_config:
+                self.ssh_params['key_filename'] = host_config['identityfile']
```
Comment (shehzaman) [inline docker/transport/sshconn.py:219]: Important bits to review

**Comment by olingerc** [comment]
Could you please merge this PR. Thanks a lot!

**Review by ulyssessouza — APPROVED** [review]
LGTM

**Review by aiordache — APPROVED** [review]

**Pull request merged**